web/src/hooks/useUserLocation.ts

```diff
@@ -40,4 +43,19 @@ const getUserLocation = async (): Promise<UserLocationType> =>
     )
   })
 
-export default getUserLocation
+const useUserLocation = (): { data: UserLocationType | null; refresh: () => void } => {
+  const [refreshState, setRefreshState] = useState(false)
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const { data } = useLoadAsync(useCallback(async () => getUserLocation(), [refreshState]))
+
+  const refresh = useCallback(() => {
+    setRefreshState(prev => !prev)
+  }, [])
+
+  return {
+    data,
+    refresh,
+  }
+}
```

Comment:
:wrench: This really complicates things, the additional state and everything should not be needed. Please just copy the hook as it was before in PoisPage and CitySelector and return everything and not just the data. That includes a `refresh` function already.